core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseSampleSourceTelemetry.java

```diff
@@ -17,6 +17,8 @@ public void reset() {
     @Override
     protected void setSampleRate(Envelope envelope) {
         Double currentSP = getSamplingPercentage();
-        envelope.setSampleRate(currentSP);
+        if (currentSP != null) {
```

Comment:
I understand that flow wouldn't change as the only thing you did is moved the check one level up. My question is if one called `reset` on something that was set to non-default value - is it expected that reset will simply do nothing? I'd expect that it will at least set the value to 100